src/pushcollector/_impl/proxy.py

```diff
@@ -50,11 +51,48 @@ class CollectorProxy(object):
     def __init__(self, delegate):
         self._delegate = delegate
 
+    def _translate_pushitem(self, pushitem):
+        def _get_checksum():
```

Comment:
The style here seems pretty odd, wouldn't it be less code and more straightforward if this was simply a variable assignment?

```
  checksums = None
  if (...):
    checksums = { ... }
```

I'm not getting an advantage from wrapping it up into a closure.

Reply:
+1